### PR TITLE
Source S3: validate CSV read options and convert options

### PR DESCRIPTION
**Pull request by davydov-d**
## What
https://github.com/airbytehq/oncall/issues/1467

## How
Add validation for CSV read options and convert options

### CODE REVIEW

**Comment by davydov-d** [comment]
:point_up: the failing test is not related. It's about the actual and expected schema mismatch (avro)

File: airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/formats/csv_parser.py [hunk 1]
```diff
@@ -51,22 +53,57 @@ def format(self) -> CsvFormat:
...
-            raise ValueError(f"{field_name} should contain 1 character only")
+    @staticmethod
+    def _validate_field(
+        format_: Mapping[str, Any], field_name: str, allow_empty: bool = False, disallow_values: Optional[Tuple[Any, ...]] = None
+    ) -> Optional[str]:
+        disallow_values = disallow_values or ()
+        field_value = format_.get(field_name)
+        if not field_value and allow_empty:
+            return
+        if len(format_.get(field_name)) != 1:
```
[the remark this answers is not in the document]
Comment (davydov-d) [reply airbyte-integrations/connectors/source-s3/source_s3/source_files_abstract/formats/csv_parser.py]: That's right, thanks. Updated

**Review by roman-yermilov-gl — APPROVED** [review]

**Pull request merged**